src/charm_state.py

```diff
@@ -143,6 +145,51 @@ def _get_supported_arch() -> ARCH:
             raise UnsupportedArchitectureError(arch=arch)
 
 
+class SSHDebugInfo(BaseModel):
+    """SSH connection information for debug workflow.
+
+    Attributes:
+        host: The SSH relay server host IP address inside the VPN.
+        port: The SSH relay server port.
+        rsa_fingerprint: The host SSH server public RSA key fingerprint.
+        ed25519_fingerprint: The host SSH server public ed25519 key fingerprint.
+    """
+
+    host: IPvAnyAddress
+    port: int = Field(0, gt=0, le=65535)
+    rsa_fingerprint: str
+    ed25519_fingerprint: str
+
+    @classmethod
+    def from_charm(cls, charm: CharmBase) -> Optional["SSHDebugInfo"]:
+        """Initialize the SSHDebugInfo from charm relation data.
+
+        Args:
+            charm: The charm instance.
+        """
+        relations = charm.model.relations[DEBUG_SSH_INTEGRATION_NAME]
+        if not relations or not (relation := relations[0]).units:
+            return None
+        target_unit = next(
+            filter(lambda unit: charm.app.name not in unit.name, iter(relation.units))
+        )
```

Comment:
What does this do please?

Reply:
This filters out "github_runner" units from the relation units so that we are left with tmate-ssh-server units only.

Reply:
Actually, the the relation units only consist of remote units (just checked). Removed the filter. Thank you :)